Sources/PushNotifications.swift

```diff
@@ -89,6 +96,54 @@ import Foundation
     }
     #endif
 
+    /**
+     Set user id.
+
+     - Parameter userId: User id.
+
+     - Precondition: You need to first authenticate the user by providing `beamsTokenProvider` in `start(instanceId:, beamsTokenProvider:)` method. `beamsTokenProvider` should not be nil.
+    */
+    /// - Tag: setUserId
+    @objc public func setUserId(_ userId: String) throws {
+        guard let beamsTokenProvider = self.beamsTokenProvider else {
+            throw UserValidationtError.beamsTokenProviderException
+        }
+
+        let persistenceService: UserPersistable = PersistenceService(service: UserDefaults(suiteName: "PushNotifications")!)
+        guard persistenceService.getUserId() == nil else {
+            throw UserValidationtError.userExists
+        }
+
+        beamsTokenProvider.fetchToken(userId: userId, completion: { (token) in
+
+            guard
+                let deviceId = Device.getDeviceId(),
+                let instanceId = Instance.getInstanceId(),
+                let url = URL(string: "https://\(instanceId).pushnotifications.pusher.com/device_api/v1/instances/\(instanceId)/devices/apns/\(deviceId)/user")
+                else {
+                    return
+            }
+
+            let networkService: PushNotificationsNetworkable = NetworkService(session: self.session)
+            networkService.setUserId(url: url, token: token, completion: { _ in
+                persistenceService.setUserId(userId: userId)
+            })
+        })
+    }
+
+    /**
+     Log out user and remove all subscriptions.
+
+     Remove user and all subscriptions set from both client and server.
+     Device is now in a fresh state, ready for new subscriptions or user being set.
+     */
+    /// - Tag: clearAllState
+    @objc public func clearAllState() {
+        let persistenceService: UserPersistable & InterestPersistable = PersistenceService(service: UserDefaults(suiteName: "PushNotifications")!)
```

Comment:
The `UserDefaults` `suitName` could be put in a constant somewhere as it's used in at least a couple of places.

Reply:
Agreed. I've actually found out that I've already created a constant for the [`suiteName`](https://github.com/pusher/push-notifications-swift/blob/master/Sources/Constants.swift#L5) but I've forgot to use it everywhere 🙈